src/oneD/StFlow.cpp

```diff
@@ -344,190 +354,259 @@ void StFlow::updateProperties(size_t jg, double* x, size_t jmin, size_t jmax)
+    size_t j1 = std::min(jmax, m_points - 2);
+    if (m_usesLambda == true) { // "axisymmetric-flow"
+        for (size_t j = j0; j <= j1; j++) { // interior points
+            // For "axisymmetric-flow", the continuity equation  propagates the
+            // mass flow rate information to the left (j+1 -> j) from the value
+            // specified at the right boundary. The lambda information propagates
+            // in the opposite direction.
+            rsd[index(c_offset_U,j)] = -(rho_u(x,j+1) - rho_u(x,j))/m_dz[j]
+                                    -(density(j+1)*V(x,j+1) + density(j)*V(x,j));
+            diag[index(c_offset_U, j)] = 0; // Algebraic constraint
+        }
+    } else if (m_isFree) { // "free-flow"
+        for (size_t j = j0; j <= j1; j++) {
+            // terms involving V are zero as V=0 by definition
+            if (grid(j) > m_zfixed) {
+                rsd[index(c_offset_U,j)] = -(rho_u(x,j) - rho_u(x,j-1))/m_dz[j-1];
+            } else if (grid(j) == m_zfixed) {
+                if (m_do_energy[j]) {
+                    rsd[index(c_offset_U,j)] = (T(x,j) - m_tfixed);
+                } else {
+                    rsd[index(c_offset_U,j)] = (rho_u(x,j) - m_rho[0]*0.3); // why 0.3?
+                }
+            } else { // grid(j < m_zfixed
+                rsd[index(c_offset_U,j)] = -(rho_u(x,j+1) - rho_u(x,j))/m_dz[j];
             }
+            diag[index(c_offset_U, j)] = 0; // Algebraic constraint
+        }
+    } else { // "unstrained-flow" (fixed mass flow rate)
+        for (size_t j = j0; j <= j1; j++) {
+            rsd[index(c_offset_U, j)] = rho_u(x, j) - rho_u(x, j - 1);
+            diag[index(c_offset_U, j)] = 0; // Algebraic constraint
+        }
+    }
+}
 
-            // The default boundary condition for species is zero flux. However,
-            // the boundary object may modify this.
+void StFlow::evalMomentum(double* x, double* rsd, int* diag,
+                          double rdt, size_t jmin, size_t jmax)
+{
+    if (m_usesLambda == false) { //disable this equation
+        for (size_t j = jmin; j <= jmax; j++) {
+            rsd[index(c_offset_V, j)] = V(x, j);
+            diag[index(c_offset_V, j)] = 0;
+        }
+        return;
+    }
+
+    if (jmin == 0) { // left boundary
+        rsd[index(c_offset_V,jmin)] = V(x,jmin);
+    }
+
+    if (jmax == m_points - 1) { // right boundary
+        rsd[index(c_offset_V,jmax)] = V(x,jmax);
+        diag[index(c_offset_V,jmax)] = 0;
+    }
+
+    // j0 and j1 are constrained to only interior points
+    size_t j0 = std::max<size_t>(jmin, 1);
+    size_t j1 = std::min(jmax, m_points - 2);
+    for (size_t j = j0; j <= j1; j++) { // interior points
+        rsd[index(c_offset_V,j)] = (shear(x, j) - lambda(x, j)
+                                   - rho_u(x, j) * dVdz(x, j)
+                                   - m_rho[j] * V(x, j) * V(x, j)) / m_rho[j]
+                                   - rdt * (V(x, j) - V_prev(j));
```

Comment:
This indentation somewhat more clearly indicates the terms that are within the parentheses:
```suggestion
        rsd[index(c_offset_V,j)] = (shear(x, j) - lambda(x, j)
                                    - rho_u(x, j) * dVdz(x, j)
                                    - m_rho[j] * V(x, j) * V(x, j)) / m_rho[j]
                                   - rdt * (V(x, j) - V_prev(j));
```

Reply:
Ah, I totally missed that.